### PR TITLE
add Workspaces Roles List

### PR DESCRIPTION
**Pull request by InsaneZein**
### Description
Add Roles list

[RHCLOUD-34813](https://issues.redhat.com/browse/RHCLOUD-34813)

---

### Screenshots

image

---

### Checklist ☑️
- [ ] PR only fixes one issue or story 
- [ ] Change reviewed for extraneous code 
- [ ] UI best practices adhered to 
- [ ] Commits squashed and meaningfully named 
- [ ] All PR checks pass locally (build, lint, test, E2E)

##
- [ ] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [ ] _(Optional) QE: Has been mentioned_
- [ ] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [ ] _(Optional) UX: Has been mentioned_
##

### CODE REVIEW

File: src/smart-components/role/RolesTable.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  const pagination = useDataViewPagination({ perPage: 20 });
+  const { page, perPage } = pagination;
+
+  const selection = useDataViewSelection({ matchOption: (a, b) => a === b });
+  const { selected, onSelect, isSelected } = selection;
+
+  const fetchData = useCallback(
+    (apiProps: { limit: number; offset: number; orderBy: string }) => {
+      const { limit, offset, orderBy } = apiProps;
+      dispatch(fetchRoles({ ...mappedProps({ limit, offset, orderBy }), usesMetaInURL: true }));
```
Comment (fhlavac) [inline src/smart-components/role/RolesTable.tsx:78]: I think we should be able to go without `usesMetaInURL` with the new URL management in DataView. It was needed when we managed that in RBAC before

File: src/smart-components/role/RolesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import { ActionsColumn } from '@patternfly/react-table';
+import ContentHeader from '@patternfly/react-component-groups/dist/esm/ContentHeader';
+import { fetchRoles } from '../../redux/actions/role-actions';
+import { useIntl } from 'react-intl';
+import messages from '../../Messages';
+import { mappedProps } from '../../helpers/shared/helpers';
+import { Role } from '../../redux/reducers/role-reducer';
+import { RBACStore } from '../../redux/store';
+
+const intl = useIntl();
```
Comment (fhlavac) [inline src/smart-components/role/RolesTable.tsx:18]: @InsaneZein here you are using the hook outside the component. It's probably the cause of the error

Comment (InsaneZein) [reply src/smart-components/role/RolesTable.tsx:18]: Thank you! I knew I missed one somewhere :sweat_smile:

**Comment by InsaneZein** [comment]
I still need to figure out the routing, but the rest is ready for review.

**Comment by fhlavac** [comment]
> the roles table can be accessed at /new-roles/ temporarily

Maybe it would be better to keep it at the original /roles route and just switch the component rendered based on the feature flag value. What do you think?

**Comment by fhlavac** [comment]
@InsaneZein looks great! 
Can you please add some simple cypress tests to check that the page is visible? Thank you 🙂

**Comment by InsaneZein** [comment]
@fhlavac I've added some cypress tests

File: src/utilities/pathnames.js
Comment (fhlavac) [inline src/utilities/pathnames.js:1]: We probably don't need this one

File: src/utilities/pathnames.js [hunk 3]
```diff
@@ -13,6 +13,11 @@ const pathnames = {
     path: '/workspaces/*',
     title: 'Workspaces',
   },
+  newRoles: {
+    link: '/new-roles',
+    path: '/new-roles/*',
+    title: 'Roles',
+  },
```
Comment (fhlavac) [inline src/utilities/pathnames.js:20]:
```suggestion
```
Just removing the old path

**Review by fhlavac — APPROVED** [review]
🎉 Looks great! Thanks @InsaneZein

**Pull request merged**